resources/sql/queries.sql

```diff
@@ -620,3 +638,15 @@ SELECT * FROM (
   ORDER BY 1 desc
   LIMIT :limit_days) AS a
 ORDER BY a.day ASC;
+
+-- :name contract-from-pool :? :1
+-- :doc return first contract from archive for given owner
+SELECT a.data->>'transaction_hash' as transaction_hash
```

Comment:
Not totally convinced about treating an archive to use application logic there apart from archiving. How about modelling the Ethereum contract in a table, even if only with the params that we need right now (I think that `transaction_hash` and `owner_address`), and we don't need to archive it, but just need an attribute for status?

Reply:
@siphiuel Let me understand well the query before elaborating the proposal. Why are we selecting `a.data->>'transaction_hash'` and then filtering for it to be null with `a.data->>'transaction_hash' = i.transaction_hash` and `i.transaction_hash is null`? I might be understanding badly the logic. :S

Reply:
We use left join condition `issues.transaction_hash=null` so that to pick transaction hashes that are not in use already. It might happen that a record has been removed from `issues` table to the `archive`, and then restored. However, the `archive` table is append-only, so this will result in the same transaction hash being present both in `issues` and in the `archive`, and we want to filter that away.

Reply:
Okay, got it, thanks!

So my proposal was about starting modelling the Ethereum contract, since it is a model with some data in the end. And doing so only with the attributes and behaviours of the contract that we need for this.

For example, we could have a table `ethereum_contract`, with attrs `id`, `owner_address`, `transaction_hash`, `status`. When an issue is archived and has a contract that we can reuse, we can create a contract model there with the status `not-used` (or maybe a better name). Then when looking for available contracts, we search there, and when taking one, we change the status of it to `used`. That way we would avoid querying the archive table for application logic, and would express more directly the status of a contract and its filtering.

Reply:
Let's handle contract modeling in a separate issue: https://github.com/status-im/open-bounty/issues/267.